### PR TITLE
Update California 2023 Individual Income Tax Parameters

### LINKED ISSUE #3769: California 2023 Individual Income Tax Parameter Updates

### PR DESCRIPTION
**Pull request by DianaShi5083**
Fixes #3769

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Update 2023 CA Income Tax Parameters
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Update 2023 California income tax parameters.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/credits/child_dependent_care/rate.yaml [hunk 2]
```diff
@@ -30,5 +30,9 @@ metadata:
       href: https://www.ftb.ca.gov/forms/2021/2021-3506-instructions.html
     - title: 2022 Instructions for California Form 3506
       href: https://www.ftb.ca.gov/forms/2022/2022-3506-instructions.html
+    - title: 2023 Instructions for California Form 3506
+      href: https://www.ftb.ca.gov/forms/2023/2023-3506-instructions.html
+    - title: 2023 California Form 3506
+    - href: https://www.ftb.ca.gov/forms/2023/2023-3506.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/credits/child_dependent_care/rate.yaml:36]: add page

File: policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/agi_fraction.yaml [hunk 3]
```diff
@@ -11,5 +11,10 @@ metadata:
     href: https://www.ftb.ca.gov/forms/2021/2021-540-ca-instructions.html
   - title: 2022 Form 540 Schedule CA instructions
     href: https://www.ftb.ca.gov/forms/2022/2022-540-ca-instructions.html
+  - title: 2023 Form 540 Schedule CA instructions, line 29
+    href: https://www.ftb.ca.gov/forms/2023/2023-540-ca-instructions.html
+  - title: 2023 Form 540 California Resident Income Tax Return, line 29
+    href: https://www.ftb.ca.gov/forms/2023/2023-540.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/agi_fraction.yaml:17]: add page - and all other .pdf links

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: could you add one integration test, where you create a test case with 2 people with employment income and calculate their standard deduction, income tax, taxable income etc. 
Calculate it from the 2023 form first to make sure we we get the right results

File: policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+- name: two people
+  period: 2023
+  input:
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml:3]:
```suggestion
  input:
     people:
        person1:
           age: 40
           employment_income: 50_000
        person2:
           age: 40
           employment_income: 40_000
    tax_units....
```

File: policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    state_code: CA
+    filing_status: JOINT
+    ca_agi: 90_000
+    ca_itemized_deductions: 10_000
+  output:
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml:8]:
```suggestion
  output:
       ca_agi: 90_000
```

File: policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+- name: two people
+  period: 2023
+  input:
+    state_code: CA
+    filing_status: JOINT
+    ca_agi: 90_000
+    ca_itemized_deductions: 10_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml:7]: keep itemized ded as an input

File: policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+- name: two people
+  period: 2023
+  input:
+    state_code: CA
+    filing_status: JOINT
+    ca_agi: 90_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ca/tax/income/integration.yaml:6]:
```suggestion
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/agi_threshold.yaml [hunk 8]
```diff
@@ -10,21 +10,29 @@ metadata:
     href: https://www.ftb.ca.gov/forms/2021/2021-540-ca-instructions.html
   - title: 2022 Form 540 Schedule CA instructions
     href: https://www.ftb.ca.gov/forms/2022/2022-540-ca-instructions.html
-  - title: Cal. Rev. & Tax. Code § 17077
+  - title: 2023 Form 540 Schedule CA instructions
+    href: https://www.ftb.ca.gov/forms/2023/2023-540-ca-instructions.html
+  - title: Cal. Rev. & Tax. Code § 17077 #the parameters in this reference didn't updated
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/agi_threshold.yaml:15]:
```suggestion
  # The legal code provides the inflation unadjusted amount
  - title: Cal. Rev. & Tax. Code § 17077 
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/exemptions/dependent_amount.yaml [hunk 9]
```diff
@@ -1,5 +1,6 @@
 description: CA dependent exemption amount
 values:
+  1999-01-01: 227
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/tax/income/exemptions/dependent_amount.yaml:3]:
```suggestion
  1999-01-01: 227 # As specified in legal code. Indexed to inflation from here.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/amount.yaml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 description: Amount to phase california exemption out by
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/amount.yaml:1]:
```suggestion
description: California reduces its exemption by this amount for each increment of income exceeding the threshold.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/start.yaml [hunk 11]
```diff
@@ -1,24 +1,29 @@
 description: Phase out starting amount of california exemption
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/start.yaml:1]:
```suggestion
description: California reduces the exemption amount for filers with income exceeding this threshold.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/increment.yaml [hunk 12]
```diff
@@ -1,14 +1,14 @@
 description: Phase out increment of california exemption
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/increment.yaml:1]:
```suggestion
description: California reduces its exemption for each of these increments of income in excess of the threshold.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/start.yaml [hunk 13]
```diff
@@ -1,24 +1,29 @@
...
 SINGLE:
+  1991-01-01: 100_000
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/tax/income/exemptions/phase_out/start.yaml:3]:
```suggestion
  1991-01-01: 100_000 # Defined in the legal code. Inflation-indexed from here.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you please:
1. Double-check the parameters I added uprating metadata to that don't have 2023 values, e.g. AMT?
2. Add remaining uprating metadata to the other thresholds as I did for `rates/head_of_household.yaml` (sorry this is annoying! should be easier post https://github.com/PolicyEngine/policyengine-core/issues/163)

File: policyengine_us/parameters/gov/states/ca/cpi.yaml [hunk 14]
```diff
@@ -7,3 +7,4 @@ values:
 metadata:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/cpi.yaml:7]:
```suggestion
  2023-01-01: 332.035
metadata:
  reference:
    title: California Consumer Price Index
    href: https://www.dir.ca.gov/oprl/CPI/EntireCCPI.PDF
```

**Comment by PavelMakarchuk** [comment]
@MaxGhenis the uprating would affect the unit tests correct?

**Comment by MaxGhenis** [comment]
Yep

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**